packages/@aws-cdk/aws-codepipeline-actions/lib/s3/deploy-action.ts

```diff
@@ -62,7 +79,205 @@ export class S3DeployAction extends Action {
         BucketName: this.props.bucket.bucketName,
         Extract: this.props.extract === false ? 'false' : 'true',
         ObjectKey: this.props.objectKey,
+        CannedACL: this.props.cannedAcl,
+        CacheControl: this.props.cacheControl && this.props.cacheControl.toString(),
       },
     };
   }
 }
+
+/**
+ * Helper class generating RFC 2616 compliant HTTP Cache-Control request or response header values
+ *
+ * @see https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9 RFC 2616 14.9 Cache-Control
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control MDN: Cache-Control
+ */
+export class CacheControlDirective {
+  protected readonly directives: string[] = [];
+
+  /**
+   * Specifies the maximum amount of time a resource will be considered fresh
+   *
+   * @param duration Maximum freshness
+   */
+  public maxAge(duration: Duration): this {
+    this.directives.push(`max-age=${duration.toSeconds()}`);
+
+    return this;
+  }
+
+  /**
+   * Forces caches to submit the request to the origin server for validation before releasing a cached copy
+   */
+  public noCache(): this {
+    this.directives.push('no-cache');
+
+    return this;
+  }
+
+  /**
+   * The cache should not store anything about the client request or server response
+   */
+  public noStore(): this {
+    this.directives.push('no-store');
+
+    return this;
+  }
+
+  /**
+   * No transformations or conversions should be made to the resource
+   * The Content-Encoding, Content-Range, Content-Type headers must not be modified by a proxy.
+   */
+  public noTransform(): this {
+    this.directives.push('no-transform');
+
+    return this;
+  }
+
+  protected constructor() {
+  }
+
+  /**
+   * Generate the string value of the built Cache-Control directive
+   */
+  public toString(): string {
+    return this.directives.join(', ');
+  }
+}
+
+/**
+ * Helper class generating RFC 2616 compliant HTTP Cache-Control request header values
+ *
+ * @see https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9 RFC 2616 14.9 Cache-Control
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Cache_request_directives MDN: Cache-Control
+ */
+export class CacheControlRequest extends CacheControlDirective {
```

Comment:
Where is this used? Seems like we only have `CacheControlResponse`

Reply:
Like I previously mentioned, I haven't found an explicit mention that Cache-Control request values shouldn't be used in S3, but I don't see a logical use case for them.

Reply:
Done